### ISSUE TITLE
(GitHub code colors) change syntax

### ISSUE THREAD

**Issue by jerone** [Mottie/GitHub-userscripts#95]
Hi Mottie. Love your [GitHub code colors](https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-code-colors) userscript!

Wondering if it would be possible to add a toggle event to the color preview for changing the color syntax. I'm imagining that everything the color preview is pressed, the color syntax changes to another syntax. For example from hex to rgba, click again to hsl, etc...

**Comment by Mottie** [comment]
Hi @jerone!

What if we instead add a tooltip with all the color syntax alternatives? That'd be way easier than messing with the DOM and including syntax highlighting.

**Comment by jerone** [comment]
I should have specified; my main reason was to copy another syntax.

**Comment by Mottie** [comment]
Ahhh, ok. Let me look into it.

**Comment by Mottie** [comment]
So, I found a nice color conversion library. I was thinking it might be nice to include other colors like hwb, cmyk, etc. What do you think?

![](https://user-images.githubusercontent.com/136959/61751952-adc73700-ad6f-11e9-8b11-d9b117cb22b4.png)

There are a bunch of other color formats, but I think that might be too much:

- hsv
- xyz
- lab
- lch
- ansi16
- ansi256
- hcg

**Comment by Mottie** [comment]
Also, if a color includes an alpha channel, it adds more definitions:

![](https://user-images.githubusercontent.com/136959/61752977-5f1b9c00-ad73-11e9-9a60-8dbe13f7c813.png)

It's probably OCD of me, but I think I should include a `;` after each color...

**Comment by Mottie** [comment]
```css
.style {
  background: #123456;
  color: rgb( 255,128,64);
  border-left: 1px hsla(65, 100%, 50%, 1) solid;
  border-right-color: maroon;
  box-shadow: 3px 3px hsl( 128, 30%, 50% );
  text-shadow: 2px 5px rgba(255, 0,  183,0.5);
}
```

**Comment by Mottie** [comment]
```css
.style {
  background: #123456;
  color: rgb( 255,128,64);
  border-left: 1px hsla(65, 100%, 50%, 1) solid;
  border-right-color: maroon;
  box-shadow: 3px 3px hsl( 128, 30%, 50% );
  text-shadow: 2px 5px rgba(255, 0,  183,0.5);
}
```

**Comment by Mottie** [comment]
I originally was using a details/summary to show the popup, but using the copy code snippet button would include *all* of the popup content. So now it's a button. The copy code snippet will still include the popup content if the popup is visible.

**Comment by jerone** [comment]
Awesome work, as always!

But I found two issues:

1. The tooltips won't close by itself. I have to press exactly at the same color preview to close the tooltip. I like that it will stay open, because then I can do the next issue (see 2.).
  ![image](https://user-images.githubusercontent.com/55841/61774675-a3e91680-adf7-11e9-8634-de9344ffa012.png)

2. It won't allow me to select any of the conversions, so I can copy it.

Tested here: https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-code-colors

<br/>
<br/>
<br/>

> It's probably OCD of me, but I think I should include a `;` after each color...

:) In the CSS world yes, but it can be used at other places too. Better to leave it as it is now (without).
